### DISCUSSION TITLE
How to create a menu with rounded corners

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#12236] [Q&A] [answered]
How can I create a Menu with rounded corners?

**Comment by danielmayost** [comment]
```
<Menu>
  <MenuItem Header="File" CornerRadius="8">
    <MenuItem Header="One" />
    <MenuItem Header="Two" />
  </MenuItem>
</Menu>
```

**Reply by wgraaf** [reply]
Unfortunately, that does not result in rounded corners for the popup with sub-menus.
Instead, it results in rounded corners for the underlying 'File'-button only:

![Screenshot 2023-07-17 144029](https://github.com/AvaloniaUI/Avalonia/assets/123667061/5b9017f2-7931-44fa-9611-4f29d915c754)

**Reply by danielmayost** [reply]
```
<Menu>
  <Menu.Resources>
    <CornerRadius x:Key="OverlayCornerRadius">15</CornerRadius>
  </Menu.Resources>
  <MenuItem Header="File">
    <MenuItem Header="One" />
    <MenuItem Header="Two" />
  </MenuItem>
</Menu>
```

**Reply by wgraaf** [reply]
No cigar:
![Screenshot 2023-07-17 150413](https://github.com/AvaloniaUI/Avalonia/assets/123667061/8057aee7-852f-4ea1-981d-bea953753708)

**Reply by danielmayost** [reply]
With Fluent Style it will work what I wrote you before, if you work with Simple style then like this:
```
<Menu>
  <Menu.Styles>
    <Style Selector="MenuItem Popup#PART_Popup Border">
      <Setter Property="CornerRadius" Value="18"/>
    </Style>
  </Menu.Styles>
  <MenuItem Header="File">
    <MenuItem Header="One" />
    <MenuItem Header="Two" />
  </MenuItem>
</Menu>
```

**Reply by wgraaf** [reply, marked as answer]
That works, thanks!
